src/java/org/apache/cassandra/index/sai/plan/VectorTopKProcessor.java

```diff
@@ -187,7 +185,6 @@ private StorageAttachedIndex findVectorIndexFor(SecondaryIndexManager sim, RowFi
         if (e.operator() != Operator.ANN)
             return null;
 
-        Optional<Index> index = sim.getBestIndexFor(e);
-        return (StorageAttachedIndex) index.filter(i -> i instanceof StorageAttachedIndex).orElse(null);
+        return sim.getBestIndexFor(e, StorageAttachedIndex.class).orElse(null);
```

Comment:
@adelapena This seems like a great simplification, I am not sure why the original implementator chose this approach. I think that the version of `getBestIndexFor` accepting class was there already when this was implemented or I am not getting something ...